Twilio-org/phonebank
# DOTORG-107: Call Submission

# Description
Calls are now being submitting in front end with responses

### Overall Changes

#### 1.) Form Validation under `/helpers`
- Prevent submit if questions are not all answered
- Prevent submit if call is not 'HUNG_UP'

#### 2.) Action under `/actions/calls.jsx`
- make PUT request to users/:id/campaigns/:campaign_id/calls/:call_id
- if outcome is not 'ANSWERED', do not pass any responses


#### 3.) Serializer Helper Function under `/helpers`
- wrote a helper function to serialize responses appropriately before sending request (I used it in the action)

#### 4.) Components
- created `RadioGroup` and `CheckboxGroup` under `components/common/form` that shows some errors if present. These components should be used with redux-form's `FieldGroup` component instead of `Field` because it contains several fields.

#### 5.) Calls Controller
- responses don't need to be parsed, changed accordingly
- changed some logic with call outcome to make it accept no responses if outcome is not 'ANSWERED'
***** the controller right now is still failing if you pass in empty responses because it tries to map empty, I left that alone because I felt that it would be even more out of scope.

## Test Plan
- Go to localhost, login, start calling on a joined campaign
- Start call and see interface handling empty questions.
